src/gen_alpha_poc/state.h

```diff
@@ -7,37 +7,51 @@ namespace openturbine::gen_alpha_solver {
 /// @brief Class to store and manage the states of a dynamic system
 class State {
 public:
-    /// Default constructor that initializes all states to zero with size one
+    static constexpr size_t kNumberOfLieGroupComponents = 7;
+    static constexpr size_t kNumberOfLieAlgebraComponents = 6;
+
+    /// Default constructor that initializes all states to zero (assuming a single node)
     State();
 
+    /// Constructor that initializes the states to the given 2D Kokkos views
     State(
-        Kokkos::View<double*> generalized_coords, Kokkos::View<double*> velocity,
-        Kokkos::View<double*> acceleration, Kokkos::View<double*> algorithmic_acceleration
+        Kokkos::View<double* [kNumberOfLieGroupComponents]> gen_coords,
+        Kokkos::View<double* [kNumberOfLieAlgebraComponents]> velocity,
+        Kokkos::View<double* [kNumberOfLieAlgebraComponents]> accln,
+        Kokkos::View<double* [kNumberOfLieAlgebraComponents]> algo_accln
     );
 
-    /// Returns the generalized coordinates vector
-    inline Kokkos::View<double*> GetGeneralizedCoordinates() const { return generalized_coords_; }
+    /// Returns the generalized coordinates vector (read only)
+    inline Kokkos::View<const double* [kNumberOfLieGroupComponents]> GetGeneralizedCoordinates(
```

Comment:
Do we want the view type to be `const double`?  Don't we want to be able to edit the data that we get through this API?